src/cli/common.rs

```diff
@@ -424,14 +424,20 @@ pub fn list_installed_components(toolchain: &Toolchain<'_>) -> Result<()> {
     Ok(())
 }
 
-pub fn list_toolchains(cfg: &Cfg, is_verbose: bool) -> Result<()> {
-    let toolchains = cfg.list_toolchains()?;
-    let toolchain_info = if is_verbose {
-        format!("\t{}", &cfg.rustup_dir.display())
+fn print_toolchain_path(cfg: &Cfg, toolchain: String, if_default: &str, verbose: bool) {
```

Comment:
It might be cute to also handle symlinked toolchains (`rustup toolchain link`) by reading the symlink and reporting the target rather than the location in `~/.rustup/toolchains`

This might be doable with (untested pseudorust):
```rust
let toolchain_path = {
    let mut t_path = cfg.toolchains_dir.clone();
    t_path.push(&toolchain);
    t_path
};
let toolchain_meta = std::fs::symlink_metadata(&toolchain_path)?;
let toolchain_path = if verbose {
    if toolchain_meta.is_dir() {
        format!("\t{}", toolchain_path.display())
    } else {
        format!("\t{}", std::fs::read_link(toolchain_path)?)
    }
} else {
    String::new()
}
```

You can test whatever you come up with, by doing something like `rustup toolchain link system /usr` and then `rustup toolchain list -v` -- I'd expect the `system` toolchain's path to be listed as `/usr` rather than `/home/whatever/.rustup/toolchains/system`